components/health_controller/src/slow_score.rs

```diff
@@ -37,26 +52,34 @@ pub struct SlowScore {
     round_ticks: u64,
     // Identify every ticks.
     last_tick_id: u64,
-    // If the last tick does not finished, it would be recorded as a timeout.
-    last_tick_finished: bool,
+    // If the tick_id is less than `last_tick_id`-`min_timeout_ticks`, we view
+    // this tick as timeout.
+    min_timeout_ticks: u64,
+    uncompleted_ticks: HashSet<u64>,
 }
 
 impl SlowScore {
-    pub fn new(inspect_interval: Duration) -> SlowScore {
+    pub fn new(
+        timeout_threshold: Duration,
+        inspect_interval: Duration,
+        ratio_thresh: f64,
+        round_ticks: u64,
+    ) -> SlowScore {
         SlowScore {
             value: OrderedFloat(1.0),
 
             timeout_requests: 0,
             total_requests: 0,
 
-            inspect_interval,
-            ratio_thresh: OrderedFloat(0.1),
+            timeout_threshold,
+            ratio_thresh: OrderedFloat(ratio_thresh),
             min_ttr: RECOVERY_INTERVALS,
             last_record_time: Instant::now(),
             last_update_time: Instant::now(),
-            round_ticks: 30,
+            round_ticks,
             last_tick_id: 0,
-            last_tick_finished: true,
+            min_timeout_ticks: timeout_threshold.div_duration_f64(inspect_interval) as u64,
```

Comment:
```suggestion
            min_timeout_ticks: timeout_threshold.div_duration_f64(inspect_interval).cell() as u64,
```
And it should also consider the corner case that `inspect_interval.is_zero()`